src/renderer/src/hooks/stores/mapeoDeviceStore.ts

```diff
@@ -0,0 +1,122 @@
+import { create } from 'zustand'
+
+type DeviceType = 'mobile' | 'desktop'
+
+export type Role = 'coordinator' | 'participant' | 'waitingInvite'
+
+type MapeoDevice =
+  | {
+      name: string
+      deviceId: string
+      deviceType: DeviceType
+      pastDeviceNames?: string[]
+      role: Extract<Role, 'waitingInvite'>
+    }
+  | {
+      name: string
+      deviceId: string
+      deviceType: DeviceType
+      pastDeviceNames?: string[]
+      role: Extract<Role, 'coordinator' | 'participant'>
+      dateAdded: Date
+    }
+
+type MapeoDeviceState = {
+  devices: Record<string, MapeoDevice>
```

Comment:
could maybe narrow if we know which `MapeoDevice` field is being used for the id e.g.

```suggestion
  devices: Record<MapeoDevice['deviceId'], MapeoDevice>
```